plugins/janus_sip.c

```diff
@@ -1383,6 +1397,28 @@ static void janus_sip_remove_quotes(char *str) {
 	}
 }
 
+static json_t *janus_sip_get_incoming_headers(const sip_t *sip, const janus_sip_session *session) {
+	json_t *headers = json_object();
+	if (!sip) {
+		return headers;
+	}
+        sip_unknown_t *unknown_header = sip->sip_unknown;
+        while (unknown_header != NULL) {
+                GList *temp = session->incoming_header_prefixes;
+                while (temp != NULL) {
+                        char *header_prefix = (char *) temp->data;
+                        if (strncmp(unknown_header->un_name, header_prefix, strlen(header_prefix)) == 0) {
+                                const char *header_name = g_strdup(unknown_header->un_name);
+                                json_object_set(headers, header_name, json_string(unknown_header->un_value));
+                                break;
+                        }
+                        temp = temp->next;
+                }
+                unknown_header = unknown_header->un_next;
+        }
+        return headers;
+}
```

Comment:
Indentation seems broken in this function: I see spaces instead of tabs. Besides, just for code style, please remove the extra space between `if`/`while` and the curly bracket.